largescaleobserved.qmd

```diff
@@ -480,38 +474,103 @@ end
+  )
+end
 ```
 
-provides an $r^2$ value very close to one, indicating an almost perfect fit.
+Reproducing @fig-memoryfootprint and @fig-timetoconverge for the RFP storage scheme as @fig-memoryfootprintrfp and @fig-timetoconvergerfp
+
+```{julia}
+#| fig-cap: "Bar plot of memory footprint of the model representation using RFP, by minimum number of ratings per user and per movie."
+#| code-fold: true
+#| label: fig-memoryfootprintrfp
+let
+  f = Figure(resolution=(800, 1000))
+  xlabel = "Minimum number of ratings per movie"
+  mc = refarray(sizespeedrfp.mc)
+  xticks = (1:7, string.(refpool(sizespeedrfp.mc)))
+  uc = refarray(sizespeedrfp.uc)
+  Legend(
+    f[1,1], lelements, llabels, ltitle;
+    orientation=:horizontal, tellwidth=false, tellheight=true,
+  )
+  barplot!(
+    Axis(f[2,1]; xticks, ylabel="Memory requirement (GiB)",),
+    mc, sizespeedrfp.modelsz; xticks, dodge=uc, color=uc,
+  )
+  barplot!(
+    Axis(f[3,1]; xticks, ylabel="Size of L[2,2] (GiB)",),
+    mc, sizespeedrfp.L22sz; xticks, dodge=uc, color=uc,
+  )
+  barplot!(
+    Axis(f[4,1]; xlabel, xticks, ylabel="L[2,2] memory fraction",),
+    mc, sizespeedrfp.L22prop; xticks, dodge=uc, color=uc,
+  )
+  f
+end
+```
 
 ```{julia}
-r²(sizemod)
+#| fig-cap: "Bar plot of function log-likelihood evaluation time using RFP, by minimum number of ratings per user and per movie using RFP storage"
+#| code-fold: true
+#| label: fig-timetoconvergerfp
+let
+  f = Figure(resolution=(800, 1000))
+  xlabel = "Minimum number of ratings per movie"
+  mc = refarray(sizespeedrfp.mc)
+  xticks = (1:7, string.(refpool(sizespeedrfp.mc)))
+  uc = refarray(sizespeedrfp.uc)
+  Legend(
+    f[1,1], lelements, llabels, ltitle;
+    orientation=:horizontal, tellwidth=false, tellheight=true,
+  )
+  barplot!(
+    Axis(f[2,1]; xticks, ylabel="Log-likelihood evaluation time (m)",),
+    mc, sizespeedrfp.evtime ./ 60; xticks, dodge=uc, color=uc,
+  )
+  barplot!(
+    Axis(f[3,1]; xticks, ylabel="Number of evaluations to convergence",),
+    mc, sizespeedrfp.nv; xticks, dodge=uc, color=uc,
+  )
+  barplot!(
+    Axis(f[4,1]; xlabel, xticks, ylabel="Time (hr) to fit the model",),
+    mc, sizespeedrfp.fittime ./ 3600; xticks, dodge=uc, color=uc,
+  )
+  f
+end
 ```
+
+The patterns in @fig-memoryfootprintrfp are similar to those in @fig-memoryfootprint but the vertical axes are quite different.
+In the most extreme case of `mc == 1` the square storage of `L[2,2]` requires over 20 GiB but the rectangular storage requires just over 10 GiB.
+This can be the difference between being able to fit the model and not being able to fit it.
+
+The time to evaluate the log-likelihood once goes from about 5 minutes for the square storage to a little over 6 minutes for the rectangular storage on the `mc == 1` models.
+Because the number of evaluations to convergence stays the same on this set of models the time to fit the model goes from roughly 2 hours (square storage) to about 2.5 hours (rectangular storage).
```

Comment:
```suggestion
Because the number of evaluations to convergence stays the same on this set of models, the time to fit the model goes from roughly 2 hours (square storage) to about 2.5 hours (rectangular storage).
```